File: AN_2/SEM_2/MPP/java/lab5/lab5/Server/src/main/java/StartRpcServer.java
import ro.mpp2024.network.utils.AbstractServer;
import ro.mpp2024.network.utils.ServerException;
import ro.mpp2024.network.utils.RpcConcurrentServer;
import ro.mpp2024.persistence.*;
import ro.mpp2024.persistence.respository.*;
import ro.mpp2024.server.ServicesImpl;
import ro.mpp2024.services.IServices;

import java.io.IOException;
import java.util.Properties;

public class StartRpcServer {
    private static int defaultPort=55555;
    public static void main(String[] args) {

        Properties serverProps = new Properties();
        try {
            serverProps.load(StartRpcServer.class.getResourceAsStream("/server.properties"));
            System.out.println("Server properties set. ");
            serverProps.list(System.out);
        } catch (IOException e) {
            System.err.println("Cannot find server.properties "+e);
            return;
        }
        ParticipantCursaRepositoryInterface participantCursaRepositoryInterface = new ParticipantCursaRepository(serverProps);
        ParticipantRepositoryInterface participantRepositoryInterface = new ParticipantRepository(serverProps);
        EchipaRepositoryInterface echipaRepositoryInterface = new EchipaRepository(serverProps);
        CursaRepositoryInterface cursaRepositoryInterface = new CursaRepository(serverProps);
        UtilizatorRepositoryInterface utilizatorRepositoryInterface = new UtilizatorRepository(serverProps);
        IServices ServerImpl = new ServicesImpl(utilizatorRepositoryInterface,echipaRepositoryInterface,participantRepositoryInterface,cursaRepositoryInterface,participantCursaRepositoryInterface);
        int ServerPort=defaultPort;
        try {
            ServerPort = Integer.parseInt(serverProps.getProperty("server.port"));
        } catch (NumberFormatException nef){
            System.err.println("Wrong  Port Number"+nef.getMessage());
            System.err.println("Using default port "+defaultPort);
        }
        System.out.println("Starting server on port: " + ServerPort);
        AbstractServer server = new RpcConcurrentServer(ServerPort, ServerImpl);
        try {
            server.start();
        } catch (ServerException e) {
            System.err.println("Error starting the server" + e.getMessage());
        }finally {
            try {
                server.stop();
            }catch(ServerException e){
                System.err.println("Error stopping server "+e.getMessage());
            }
        }
    }
}

File: AN_2/SEM_2/MPP/c#/lab7/Server/StartRpcServer.cs
using Microsoft.Extensions.Configuration;
using Model.mpp.model;
using Networking.mpp.networking.utils;
using Persistence.mpp.persistence;
using Persistence.mpp.persistence.repository;
using Server.mpp.server;
using Services.mpp.services;

namespace Server
{
    public class StartRpcServer
    {
        private static int defaultPort = 55555;

        public static void Main(string[] args)
        {
            var config = new ConfigurationBuilder()
                .SetBasePath(Directory.GetCurrentDirectory())
                .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
                .Build();

            int serverPort = config.GetSection("Server").GetValue<int>("Port", defaultPort);

            try
            {
                var testConn = DatabaseUtils.getConnection();
                Console.WriteLine("Conexiune DB reușită: " + testConn.DataSource);
            }
            catch (Exception e)
            {
                Console.WriteLine("Eroare conexiune DB: " + e.Message);
            }

            
            IParticipantCursaRepository participantCursaRepo = new ParticipantCursaRepository();
            IParticipantRepository participantRepo = new ParticipantRepository();
            IEchipaRepository echipaRepo = new EchipaRepository();
            ICursaRepository cursaRepo = new CursaRepository();
            IUtilizatorRepository utilizatorRepo = new UtilizatorRepository();
            IServices service = new ServicesImpl(utilizatorRepo, echipaRepo, participantRepo, cursaRepo, participantCursaRepo);
            
            Console.WriteLine("Starting server on port: " + serverPort);

            AbstractServer server = new RpcConcurrentServer(serverPort, service);
            try
            {
                server.Start();
            }
            catch (Exception e)
            {
                Console.Error.WriteLine("Error starting the server: " + e.Message);
            }
            finally
            {
                try
                {
                    server.Stop();
                }
                catch (Exception e)
                {
                    Console.Error.WriteLine("Error stopping the server: " + e.Message);
                }
            }
        }
    }
}